lib/io/QuadSinkHdt.ts

```diff
@@ -1,85 +1,123 @@
-import type { WriteStream } from 'node:fs';
-import { createWriteStream } from 'node:fs';
-import * as fs from 'node:fs/promises';
-import * as Path from 'node:path';
-import * as readline from 'node:readline';
-import type * as RDF from '@rdfjs/types';
+import { unlink } from 'node:fs/promises';
+import { basename, resolve, dirname } from 'node:path';
 import * as Docker from 'dockerode';
-import type { IQuadSinkFileOptions } from './QuadSinkFile';
-import { QuadSinkFile } from './QuadSinkFile';
-import { transformToHdt, pullHdtCppDockerImage } from './rfdhdtDockerUtil';
+import { QuadSinkFile, type IQuadSinkFileOptions } from './QuadSinkFile';
+
+const HDTCPP_MOUNT_PATH = '/tmp/convert';
+const HDTCPP_DOCKER_IMAGE = 'rdfhdt/hdt-cpp:latest';
+const HDTCPP_FORMATS = new Map<string, string>([
+  [ 'application/n-quads', 'nquad' ],
+  [ 'application/n-triples', 'ntriples' ],
+  [ 'text/turtle', 'turtle' ],
+  [ 'application/rdf+xml', 'rdfxml' ],
+  [ 'text/n3', 'n3' ],
+]);
 
 export class QuadSinkHdt extends QuadSinkFile {
-  private readonly files: Set<string> = new Set();
-  private readonly deleteSourceFiles: boolean;
-  private readonly errorFileDockerRfdhdt: WriteStream;
-  private readonly poolSize: number;
+  private readonly generateIndexes: boolean;
+  private readonly conversionProgress: boolean;
+  private readonly removeSourceFiles: boolean;
+  private readonly conversionConcurrency: number;
+
+  private readonly docker: Docker;
+  private readonly filesToConvert: Set<string>;
 
-  public constructor(
-    options: IQuadSinkFileOptions,
-    poolSize = 1,
-    deleteSourceFiles = false,
-    errorFileDockerRfdhdt = './error_log_docker_rfdhdt.txt',
-  ) {
+  public constructor(options: IQuadSinkHdtOptions) {
     super(options);
-    this.deleteSourceFiles = deleteSourceFiles;
-    this.errorFileDockerRfdhdt = createWriteStream(errorFileDockerRfdhdt);
-    this.poolSize = poolSize;
+    if (!HDTCPP_FORMATS.has(options.outputFormat)) {
+      throw new Error(`Unsupported HDT output format ${options.outputFormat}`);
+    }
+    this.generateIndexes = options.generateIndexes ?? false;
```

Comment:
I think it should be `true` by default because I think the query engine needs it to execute queries over them and without it, it has to generate it internally but I might be wrong @rubensworks probably knows best.

Reply:
I changed this to be true by default.